src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,979 @@
+
+            return _m_fs.Seek(offset, origin);
+        }
+
+        public override void SetLength(long value)
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            _m_fs.SetLength(value);
+        }
+
+        public override int Read([In, Out] byte[] buffer, int offset, int count)
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            return _m_fs.Read(buffer, offset, count);
+        }
+
+        public override int ReadByte()
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            return _m_fs.ReadByte();
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            _m_fs.Write(buffer, offset, count);
+
+            // SQLBUVSTS# 193123 - disable lazy flushing of written data in order to prevent
+            // potential exceptions during Close/Finalization. Since System.IO.FileStream will
+            // not allow for a zero byte buffer, we'll create a one byte buffer which, in normal
+            // usage, will cause System.IO.FileStream to utilize the user-supplied buffer and
+            // automatically flush the data directly to the disk cache. In pathological scenarios 
+            // where the user is writing a single byte at a time, we'll explicitly call flush ourselves.
+            if (count == 1)
+            {
+                // calling flush here will mimic the internal control flow of System.IO.FileStream
+                _m_fs.Flush();
+            }
+        }
+
+        public override void WriteByte(byte value)
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            _m_fs.WriteByte(value);
+
+            // SQLBUVSTS# 193123 - disable lazy flushing of written data in order to prevent
+            // potential exceptions during Close/Finalization. Since our internal buffer is
+            // only a single byte in length, the provided user data will always be cached.
+            // As a result, we need to be sure to flush the data to disk ourselves.
+
+            // calling flush here will mimic the internal control flow of System.IO.FileStream
+            _m_fs.Flush();
+        }
+
+        #endregion
+
+        static private readonly char[] s_invalidPathChars = Path.GetInvalidPathChars();
+
+        // path length limitations:
+        // 1. path length storage (in bytes) in UNICODE_STRING is limited to UInt16.MaxValue bytes = Int16.MaxValue chars
+        // 2. GetFullPathName API of kernel32 does not accept paths with length (in chars) greater than 32766
+        //  (32766 is actually Int16.MaxValue - 1, while (-1) is for NULL termination)
+        // We must check for the lowest value between the the two
+        private const int MaxWin32PathLength = short.MaxValue - 1;
```

Comment:
For the record, this isn't technically correct, but is unlikely to cause grief. `\??\UNC\` gets turned into `\Device\Mup\` under the covers. Null termination isn't a thing either once you're in UNICODE_STRING format.

Ideally you'd just let Windows choke and deal with the returned error rather than try to second guess here.

Reply:
Getting rid of this check.